### PR TITLE
Feature/framework support authenticated admin api

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### Fixes #. add authentication support for SG admin APIs

- [X] Ran `flake8`
- [X] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- add authentication support for SG admin APIs

### CODE REVIEW

File: testsuites/CBLTester/CBL_Functional_tests/conftest.py [hunk 1]
```diff
@@ -487,6 +487,11 @@ def params_from_base_suite_setup(request):
         expected_sync_gateway_version=sync_gateway_version
     )
 
+    need_sgw_admin_auth = (not disable_admin_auth) and sync_gateway_version >= "3.0"
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/conftest.py:491]: hope you have used for debugging, you want to clean up?

File: testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py [hunk 2]
```diff
@@ -69,6 +69,9 @@ def params_from_base_suite_setup(request):
     disable_tls_server = request.config.getoption("--disable-tls-server")
 
     disable_admin_auth = request.config.getoption("--disable-admin-auth")
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py:72]: clean up prints

File: testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py [hunk 3]
```diff
@@ -248,6 +251,12 @@ def params_from_base_suite_setup(request):
         expected_server_version=server_version,
         expected_sync_gateway_version=sync_gateway_version
     )
+
+    need_sgw_admin_auth = (not disable_admin_auth) and sync_gateway_version >= "3.0"
```
Comment (sridevi-15) [inline testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py:255]: Cleanup prints

File: testsuites/CBLTester/topology_specific_tests/sync_gateway_proxy_enabled/conftest.py [hunk 4]
```diff
@@ -253,6 +256,11 @@ def params_from_base_suite_setup(request):
         expected_sync_gateway_version=sync_gateway_version
     )
 
+    need_sgw_admin_auth = (not disable_admin_auth) and sync_gateway_version >= "3.0"
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/CBLTester/topology_specific_tests/sync_gateway_proxy_enabled/conftest.py:260]: Clean up prints

File: testsuites/CBLTester/upgrade_tests/conftest.py [hunk 5]
```diff
@@ -388,6 +388,10 @@ def params_from_base_suite_setup(request):
     if sync_gateway_version < "2.0":
         pytest.skip('Does not work with sg < 2.0 , so skipping the test')
 
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/conftest.py:391]: Clean up prints

File: testsuites/CBLTester/upgrade_tests/conftest.py [hunk 6]
```diff
@@ -436,6 +440,11 @@ def params_from_base_suite_setup(request):
 
     utils_obj = Utils(base_url)
 
+    need_sgw_admin_auth = (not disable_admin_auth) and sync_gateway_version >= "3.0"
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/conftest.py:444]: Clean up prints

File: testsuites/syncgateway/functional/tests/conftest.py [hunk 7]
```diff
@@ -237,6 +237,9 @@ def params_from_base_suite_setup(request):
     enable_server_tls_skip_verify = request.config.getoption("--enable-server-tls-skip-verify")
 
     disable_admin_auth = request.config.getoption("--disable-admin-auth")
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/syncgateway/functional/tests/conftest.py:240]: Clean up prints

File: testsuites/syncgateway/functional/tests/conftest.py [hunk 8]
```diff
@@ -480,6 +483,11 @@ def params_from_base_suite_setup(request):
         expected_sync_gateway_version=sync_gateway_version
     )
 
+    need_sgw_admin_auth = (not disable_admin_auth) and sync_gateway_version >= "3.0"
+    print("==============================================")
```
Comment (sridevi-15) [inline testsuites/syncgateway/functional/tests/conftest.py:487]: Clean up prints

**Review by sridevi-15 — CHANGES_REQUESTED** [review]
have a quesiton on hard coded username and password.
Remove prints which you added for debugging.

**Review by hthomps-cb — APPROVED** [review]

File: testsuites/CBLTester/CBL_Functional_tests/conftest.py (hunk 1 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/conftest.py]: cleaned up the print statement, and added proper log_info

File: testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py (hunk 2 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py]: cleaned up the print statement, and added proper log_info

File: testsuites/CBLTester/topology_specific_tests/sync_gateway_proxy_enabled/conftest.py (hunk 4 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/topology_specific_tests/sync_gateway_proxy_enabled/conftest.py]: cleaned up the print statement, and added proper log_info

File: testsuites/CBLTester/upgrade_tests/conftest.py (hunk 6 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/upgrade_tests/conftest.py]: cleaned up the print statement, and added proper log_info

File: testsuites/syncgateway/functional/tests/conftest.py (hunk 8 above)
Comment (euniceatcouchbase) [reply testsuites/syncgateway/functional/tests/conftest.py]: cleaned up the print statement, and added proper log_info

File: testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py (hunk 3 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/topology_specific_tests/multiple_sync_gateways/conftest.py:253]: cleaned up the print statement, and added proper log_info

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_channel_revocation.py [hunk 9]
```diff
@@ -80,19 +82,20 @@ def test_user_removed_from_channel_basic(params_from_base_test_setup, replicator
     channels_sg = ["A", "B"]
     username = "autotest"
     password = "password"
-    sg_client.create_user(sg_admin_url, sg_db, username, password, channels=channels_sg)
-    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
-    session = cookie, session_id
+    auth = need_sgw_admin_auth and (RBAC_FULL_ADMIN['user'], RBAC_FULL_ADMIN['pwd']) or None
```
[the remark this answers is not in the document]
Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_channel_revocation.py:85]: I defined a Full Admin RBAC user in constants.py, this new user will be created at CBS provisioning
https://github.com/couchbaselabs/mobile-testkit/pull/1917/files#diff-001178355b7ed36f1765728b56e429349faddfdd1c21d82d6f7f01721cdb6168

**Review by sridevi-15 — APPROVED** [review]

**Pull request merged**